src/query/storages/stage/src/input_context.rs

```diff
@@ -0,0 +1,145 @@
+// Copyright 2021 Datafuse Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::io::Read;
+use std::sync::Arc;
+
+use dashmap::DashMap;
+use databend_common_catalog::plan::DataSourcePlan;
+use databend_common_catalog::plan::PartInfo;
+use databend_common_catalog::plan::PartStatistics;
+use databend_common_catalog::plan::Partitions;
+use databend_common_catalog::plan::PartitionsShuffleKind;
+use databend_common_catalog::plan::Projection;
+use databend_common_catalog::plan::PushDownInfo;
+use databend_common_catalog::plan::StageTableInfo;
+use databend_common_catalog::table_context::TableContext;
+use databend_common_exception::Result;
+use databend_common_expression::TableSchemaRefExt;
+use databend_common_pipeline_core::Pipeline;
+use databend_common_pipeline_sources::input_formats::InputContext;
+use databend_common_pipeline_sources::input_formats::SplitInfo;
+use databend_common_storage::STDIN_FD;
+use log::debug;
+use opendal::Scheme;
+
+use crate::StageTable;
+
+impl StageTable {
+    pub async fn read_partition_old(
```

Comment:
The filename doesn't quite fit. The function is used to construct `Partitions`, but the filename is `input_context.rs`.

Reply:
this file extract the temporary bridge to the old impl based on InputContext.
I renamed it to `input_context_bridge`.